sfxclient/gometrics.go

```diff
@@ -12,14 +12,34 @@ var startTime = time.Now()
 // GoMetricsSource is a singleton Collector that collects basic go system stats.  It currently
 // collects from runtime.ReadMemStats and adds a few extra metrics like uptime of the process
 // and other runtime package functions.
-var GoMetricsSource Collector = &goMetrics{}
+var GoMetricsSource MetricSource = &goMetrics{}
 
-type goMetrics struct{}
+// MetricSource interface implements the necessary api's for goMetric
+type MetricSource interface {
+	Collector
+	AddCallback(func(stats runtime.MemStats))
+}
+
+// goMetrics stores process runtime memory stats
+type goMetrics struct {
+	cb func(stats runtime.MemStats)
+}
+
+// AddCallback register func which needs to be called everytime stats are collected
+func (g *goMetrics) AddCallback(f func(stats runtime.MemStats)) {
+	g.cb = f
```

Comment:
addCallback should be additional, so more than one can be called. make cb a slice.

Reply:
ack